app/validators/flood_risk_engine/grid_reference_validator.rb

```diff
@@ -0,0 +1,52 @@
+# Validate that a grid reference can be processed.
+module FloodRiskEngine
+  class GridReferenceValidator < ActiveModel::EachValidator
```

Comment:
There is a requirement for 10 number grid references in the project. Also spaces between the groups of letters and numbers are valid, handled by OsMapRef, so I think we might as well use them.